### PR TITLE
BZ 2011 Create oic.wk.mnon-schema.json

### PR DESCRIPTION
**Pull request by WAvdBeek**

### CODE REVIEW

File: schemas/oic.wk.mnon-schema.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
+{
+  "id": "http://openinterconnect.org/schemas/oic.wk.nmon#",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description" : "Copyright (c) 2017 Open Connectivity Foundation, Inc. All rights reserved.",
+  "title": "network monitoring",
+  "definitions": {
+    "nmon": {
+        "network":
```
Comment (WAvdBeek) [inline schemas/oic.wk.mnon-schema.json]: rename to iana-interfacetype.

https://www.iana.org/assignments/ianaiftype-mib/ianaiftype-mib

Comment (mkettrick) [inline schemas/oic.wk.mnon-schema.json]: I don't see this Property name in the CR.

File: schemas/oic.wk.mnon-schema.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        }
+    }
+  },
+  "type": "object",
+  "allOf": [
+    {"$ref": "oic.core.json#/definitions/oic.core"},
+    {"$ref": "oic.baseResource.json#/definitions/oic.r.baseresource"},
+    {"$ref": "#/definitions/nmon"}
+  ],
+  "required": ["reset", "col"]
```
Comment (mkettrick) [inline schemas/oic.wk.mnon-schema.json]: The CR says that "ianaifType" is a required Property as well.

File: schemas/oic.wk.nmon-schema.json [hunk 3]
```diff
@@ -0,0 +1,78 @@
+{
+  "id": "http://openinterconnect.org/schemas/oic.wk.nmon#",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description" : "Copyright (c) 2017 Open Connectivity Foundation, Inc. All rights reserved.",
+  "title": "network monitoring",
+  "definitions": {
+    "nmon": {
```
Comment (mkettrick) [inline schemas/oic.wk.nmon-schema.json]: I see my confusion.  On every other schema file I've ever looked at, they look like this:

```
   "definitions":` {
      "oic.wk.d": {
        "type": "object",
            "properties": {
```

So they use the keyword "properties" in all cases.  Is there a reason why in your schemas you're using words like "nmon" or "powersourcedefinition"?

Then at the end, it looks like this:

 { "$ref": "#/definitions/oic.wk.d" }

So it uses the "rt" value to reference back to itself.  Can we be consistent with all of our other schema definitions with this and the power sources schema?

File: schemas/oic.wk.nmon-schema.json [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            "type" : "integer",
+            "description": "Average received message size in bytes (rx) in the collection period",
+            "readOnly" : true
+        }
+      }
+    }
+  },
+  "type": "object",
+  "allOf": [
+    {"$ref": "oic.core.json#/definitions/oic.core"},
```
Comment (WAvdBeek) [inline schemas/oic.wk.nmon-schema.json:76]: oic.core.json ==> oic.core-schema.json

**Comment by WAvdBeek** [comment]
replaced by:
https://github.com/openconnectivityfoundation/core/pull/118

**Pull request closed without merging**